infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsnext/KubernetesWorkspaceNextApplierTest.java

```diff
@@ -168,6 +168,87 @@ public void addsTwoServersForContainers() throws Exception {
             80, "test-port", emptyMap(), true, 8090, "another-test-port", emptyMap(), false));
   }
 
+  @Test
+  public void addsMachineWithServersThatUseSamePortButDifferentNames() throws Exception {
+    ChePlugin chePlugin = testChePlugin();
+    addPortToSingleContainerPlugin(chePlugin, 80, "test-port/http", emptyMap(), true);
+    addPortToSingleContainerPlugin(chePlugin, 80, "test-port/ws", emptyMap(), true);
+    applier.apply(internalEnvironment, singletonList(chePlugin));
+
+    InternalMachineConfig machineConfig = getOneAndOnlyMachine(internalEnvironment);
+    assertEquals(
+        machineConfig.getServers(),
+        expectedTwoServers(
+            80, "test-port/http", emptyMap(), true, 80, "test-port/ws", emptyMap(), true));
+  }
+
+  @Test
+  public void addsMachineWithServersThatSetProtocolAndPath() throws Exception {
+    ChePlugin chePlugin = testChePlugin();
+    addPortToSingleContainerPlugin(
+        chePlugin,
+        443,
+        "test-port",
+        ImmutableMap.of("path", "/path/1", "protocol", "https", "attr1", "value1"),
+        true);
+    applier.apply(internalEnvironment, singletonList(chePlugin));
+
+    InternalMachineConfig machineConfig = getOneAndOnlyMachine(internalEnvironment);
+    assertEquals(
+        machineConfig.getServers(),
+        expectedSingleServer(
+            443, "test-port", singletonMap("attr1", "value1"), true, "https", "/path/1"));
+  }
+
+  @Test
+  public void setsDefaultMemoryLimitForMachineAssociatedWithContainer() throws Exception {
+    applier.apply(internalEnvironment, singletonList(testChePlugin()));
+
+    InternalMachineConfig machineConfig = getOneAndOnlyMachine(internalEnvironment);
+    String memoryLimitAttribute = machineConfig.getAttributes().get(MEMORY_LIMIT_ATTRIBUTE);
+    assertEquals(memoryLimitAttribute, Integer.toString(MEMORY_LIMIT_MB * 1024 * 1024));
+  }
+
+  private ChePlugin testChePlugin() {
```

Comment:
In my opinion, `testChePlugin` sounds like it should test ChePlugin, but in facts, it creates test ChePlugin object. It does not make much sense since it is test sources but maybe just `createChePlugin` would be clearer.

Reply:
ok, will change it

Reply:
done